### PR TITLE
chore: Merge main branch into modular-sdk

### PR DESCRIPTION
**Pull request by lahirumaramba**
- Merged main branch into modular-sdk
- Moved new types added in #1220 from `auth/index.ts` to `src/auth/auth-config.ts`
- Moved new types added in #1294  from `auth/index.ts` to `src/auth/auth-config.ts`
- Updated import paths in app check and `base-auth.ts`

### CODE REVIEW

**Comment by lahirumaramba** [comment]
@hiranya911 I checked to see if the new changes are included and the new types are added to the correct files in the modular SDK. Could use another set of eyes to confirm. 😃 
Thank you!

File: etc/firebase-admin.auth.api.md [hunk 1]
```diff
@@ -75,13 +71,12 @@ export abstract class BaseAuth {
 }
 
 // @public
-export interface CreateMultiFactorInfoRequest {
-    displayName?: string;
-    factorId: string;
-}
+export type CreateMultiFactorInfoRequest = CreatePhoneMultiFactorInfoRequest;
 
+// Warning: (ae-forgotten-export) The symbol "BaseCreateMultiFactorInfoRequest" needs to be exported by the entry point index.d.ts
```
Comment (hiranya911) [inline etc/firebase-admin.auth.api.md:76]: I think these warnings ought to be resolved.

File: src/auth/token-verifier.ts [hunk 2]
```diff
@@ -370,7 +370,7 @@ export class FirebaseTokenVerifier {
     fullDecodedToken: DecodedToken,
     projectId: string | null,
     isEmulator: boolean): void {
-
+
```
Comment (hiranya911) [inline src/auth/token-verifier.ts:373]: Nit: Can remove the extra whitespace added here and below.

**Review by hiranya911 — CHANGES_REQUESTED** [review]
Looks pretty good. Just found a few Auth types that need to be exported from the entrypoint.

**Comment by lahirumaramba** [comment]
> Looks pretty good. Just found a few Auth types that need to be exported from the entrypoint.

Thanks @hiranya911 ! Made the changes, but having issues getting the CI to run... I will keep trying.

**Review by hiranya911 — APPROVED** [review]
Thanks. LGTM 👍

**Pull request merged**